flashinfer/moe_ep/kernel_src/sm120/swapab_cutedsl_megakernel/src/moe_mxfp8_glu/kernel_mxfp8_glu_fc12.py

```
@@ -0,0 +1,2336 @@
+                "v1 only implements force_static_sched=True (lean 7-warp). "
+                "Dynamic CLC (force_static_sched=False) is future work."
+            )
+
+        # Validate (ab_dtype, sf_vec_size) pairing.
+        if sf_vec_size in self.VALID_AB_DTYPE_SF_SIZE:
+            valid_ab = self.VALID_AB_DTYPE_SF_SIZE[sf_vec_size]
+            if ab_dtype not in valid_ab:
+                raise ValueError(
+                    f"ab_dtype={ab_dtype.__name__} is not valid for "
+                    f"sf_vec_size={sf_vec_size}. "
+                    f"Expected one of: {[t.__name__ for t in valid_ab_tuple]}."
```

Ruff (0.16.1)

[error] Undefined name `valid_ab_tuple`

(F821)